prov/sm2/src/sm2_coordination.c

```diff
@@ -205,20 +205,20 @@ ssize_t sm2_file_open_or_create(struct sm2_mmap *map_shared)
 			"Unable to determine page size %ld\n", page_size);
 		return -FI_EINVAL;
 	}
-
-	sprintf(template, "%s/fi_sm2_pid%d_XXXXXX", SM2_COORDINATION_DIR,
+	sprintf(template, "%s/fi_sm2_pid%7d_XXXXXX", SM2_COORDINATION_DIR,
 		getpid());
 
 	/* Assume we are the first process here.
 	 * Try to open a tmpfile file in the shm directory
 	 */
 	fd = mkostemp(template, O_RDWR);
-	assert(fd > 0);
```

Comment:
Shouldn't we check fd immediately upon return?

Reply:
we could, or we could let mmap will fail on an invalid fd, and return then.

Reply:
addressed.